src/FsLex/fslexast.fs

```diff
@@ -74,87 +72,87 @@ let numLowUnicodeChars = 128
          System.Convert.ToChar x
-         
-         
+
+
 
 let NumSpecificUnicodeChars() = specificUnicodeChars.Count
-let GetSpecificUnicodeChars() = 
-    specificUnicodeChars 
-        |> Seq.sortBy (fun (KeyValue(k,v)) -> v) 
-        |> Seq.map (fun (KeyValue(k,v)) -> k) 
-
-let GetSingleCharAlphabet() = 
-    if !unicode 
-    then Set.ofList [ for c in 0..numLowUnicodeChars-1 do yield (char c)
-                      for c in GetSpecificUnicodeChars() do yield c ]
-    else Set.ofList [ for x in 0..255 ->  (char x) ]
-         
-let GetAlphabet() = 
-    if !unicode 
+let GetSpecificUnicodeChars() =
+    specificUnicodeChars
+        |> Seq.sortBy (fun (KeyValue(k,v)) -> v)
+        |> Seq.map (fun (KeyValue(k,v)) -> k)
+
+let GetSingleCharAlphabet() =
+    if !unicode
+    then Set.ofList [ yield! { char 0 .. char <| numLowUnicodeChars-1 }
+                      yield! GetSpecificUnicodeChars() ]
+    else Set.ofList [ char 0 .. char 255 ]
+
+let GetAlphabet() =
+    if !unicode
     then Set.ofList [ for c in GetSingleCharAlphabet() do yield EncodeChar c
                       for uc in 0 .. NumUnicodeCategories-1 do yield EncodeUnicodeCategoryIndex uc ]
-    else Set.ofList [ for c in GetSingleCharAlphabet() do yield EncodeChar c ]
+    else GetSingleCharAlphabet() |> Seq.map EncodeChar |> set
+
 
-         
 //let DecodeAlphabet (x:Alphabet) = System.Convert.ToChar(x)
 
 (*
-for i in 0 .. 65535 do 
+for i in 0 .. 65535 do
     let c = char i
-    if System.Char.GetUnicodeCategory c = System.Globalization.UnicodeCategory.PrivateUse then 
+    if System.Char.GetUnicodeCategory c = System.Globalization.UnicodeCategory.PrivateUse then
         printfn "i = %x" i
 *)
 
-type Spec = 
-    { TopCode: Code;
-      Macros: (Ident * Regexp) list;
-      Rules: (Ident * Ident list * Clause list) list;
-      BottomCode: Code }
-and Clause = Regexp * Code
-and Regexp = 
+type Input =
+  | Alphabet of Alphabet
+  | UnicodeCategory of string
+  | Any
+  | NotCharSet of Set<Alphabet>
+type Regexp =
   | Alt of Regexp list
   | Seq of Regexp list
   | Inp of Input
   | Star of Regexp
   | Macro of Ident
-and Input =
-  | Alphabet of Alphabet
-  | UnicodeCategory of string 
-  | Any 
-  | NotCharSet of Set<Alphabet>
+type Clause = Regexp * Code
+type Spec =
+    { TopCode: Code
+      Macros: (Ident * Regexp) list
+      Rules: (Ident * Ident list * Clause list) list
+      BottomCode: Code }
 
-type NodeId = int   
+type NodeId = int
 
-type NfaNode = 
-    { Id: NodeId;
-      Name: string;
-      Transitions: Dictionary<Alphabet, NfaNode list>;
+type NfaNode =
+    { Id: NodeId
+      Name: string
+      Transitions: Dictionary<Alphabet, NfaNode list>
       Accepted: (int * int) list }
 
-type DfaNode = 
-    { Id: int;
-      Name: string;
-      mutable Transitions: (Alphabet * DfaNode) list;
+type DfaNode =
+    { Id: int
+      Name: string
```

Comment:
Why do DFA nodes have _names_? Aren't they supposed to be called "State 1", "State 2" etc.?